cmd/xdpcap/filter_test.go

```diff
@@ -38,7 +38,7 @@ func TestMain(m *testing.M) {
 }
 
 func TestMissingFilter(t *testing.T) {
-	_, err := newFilterWithMap(hookMap(t, 1), testOpts())
+	_, err := newFilterWithMap(hookMap(t, 1), testOpts(), false)
```

Comment:
Can we add a test for `BPF_F_XDP_HAS_FRAGS`?

If we follow the current pattern, we'll at least check we can load with that flag. It won't check we can attach to a program that has the flag itself (we'd have to rework the tests to create and use that program), but maybe that's good enough for now.

Reply:
I think this will require a bit of work, to get a dummy program to be attached in legacy and multi-buffer mode so that we can force the filter to also choose between the correct entrypoint. I'm working on a test that runs in a namespace and attaches a simple program that just XDP_PASSes all packets

Reply:
I don't think we need to attach the program: we can just create a "dummy" program, load it, and PROG_TEST_RUN it. It should tail-call into the expected xdpcap program.

What would attaching it to an interface bring us?

Reply:
yeah I modified this to use a dummy ELF and `BPF_PROG_TEST_RUN` to test both normal and mult-buffer attaching of the program